### PR TITLE
hidden secrete message in workflow YAML

### PR DESCRIPTION
**Pull request by Yancey0623**
This PR makes `OSS_SK` to be a Kubernetes Secret object and add `dry_run` argument in `couler.secret` function.

### CODE REVIEW

File: pkg/sql/codegen/couler/template.go [hunk 1]
```diff
@@ -33,26 +33,34 @@ type coulerFiller struct {
...
 
 step_envs = dict()
 {{range $k, $v := .StepEnvs}}
 step_envs["{{$k}}"] = '''{{$v}}'''
 {{end}}
 
+sqlflow_secret = None
+if "{{ .OSSSK }}" != "":
+	# note(yancey1989): set dry_run to true, just reference the secret meta to generate workflow YAML,
+	# we should create the secrete before launching sqlflowserver
```
Comment (weiguoz) [inline pkg/sql/codegen/couler/template.go]: typo: secrete

File: python/couler/couler/argo.py [hunk 2]
```diff
@@ -527,7 +527,8 @@ def _dump_yaml():
     if len(_secrets) > 0:
```
Comment (weiguoz) [inline python/couler/couler/argo.py:527]: Does `len(_secrets)` leaders to an Exception when `dry_run` is `True`?
https://github.com/sql-machine-learning/sqlflow/blob/3e11d03c0694b75a6cd8e34ac52847931bf3a647/python/couler/couler/argo.py#L741-L743

File: pkg/sql/codegen/couler/codegen.go [hunk 3]
```diff
@@ -81,6 +82,8 @@ func GenCode(programIR []ir.SQLFlowStmt, session *pb.Session) (string, error) {
 		DataSource:  session.DbConnStr,
 		StepEnvs:    stepEnvs,
 		WorkflowTTL: workflowTTL,
+		OSSSK:       os.Getenv("SQLFLOW_OSS_SK"),
+		SecretName:  os.Getenv("SQLFLOW_WORKFLOW_SECRET_NAME"),
```
Comment (typhoonzero) [inline pkg/sql/codegen/couler/codegen.go]: `SQLFLOW_WORKFLOW_SECRET_NAME` seems only used for the OSS key, maybe the k8s resource name can be a constant.

File: python/couler/couler/argo.py (hunk 2 above)
Comment (Yancey0623) [reply python/couler/couler/argo.py:527]: May not, `_secrets` would be initialized as  an empty dict:

https://github.com/sql-machine-learning/sqlflow/blob/3e11d03c0694b75a6cd8e34ac52847931bf3a647/python/couler/couler/argo.py#L23

File: pkg/sql/codegen/couler/codegen.go (hunk 3 above)
Comment (Yancey0623) [reply pkg/sql/codegen/couler/codegen.go]: Good point, I use a json string to configure the secret object.

**Review by typhoonzero — APPROVED** [review]

**Pull request merged**